### ISSUE TITLE
Benchmark: `vitest bench --run` hangs, unable to use `--reporter=hanging-process` to debug

### ISSUE THREAD

**Issue by busticated** [vitest-dev/vitest#5335]
### Describe the bug

Attempting to get up and running using the benchmark feature, i'm hitting a hanging process error but cannot debug it sufficiently. 

**Repro Steps**
1. create a new file: `./src/hello.benchmark.ts` 
2. populate that file with the example benchmark ([docs](https://vitest.dev/api/#bench)):

```ts
import { bench } from 'vitest';


describe('Viewer Libs Benchmarks', () => {
	bench('normal sorting', () => {
		const x = [1, 5, 4, 2, 3];
		x.sort((a, b) => {
			return a - b;
		});
	});
});

```

3. run the benchmark: `vitest bench --run`
4. observe the run hangs and ultimately outputs:

```
close timed out after 10000ms
Tests closed successfully but something prevents Vite server from exiting
You can try to identify the cause by enabling "hanging-process" reporter. See https://vitest.dev/config/#reporters
```

5. re-run the benchmark using the built-in ([docs]()https://vitest.dev/guide/reporters.html#hanging-process-reporter) "hanging-process" reporter: `vitest bench --run --reporter=hanging-process`
6. observe error output:

```
Error: Failed to load custom Reporter from hanging-process
```

**Expected**
Benchmark runs and exits without issue

**Actual**
Benchmark hangs for some reason before being stopped by the default timeout

### Reproduction

i think my steps are pretty straight forward but i can also try to put together a live demo when i get a moment :+1:

### System Info

```shell
System:
    OS: macOS 14.1
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    Memory: 1.97 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/Library/Caches/fnm_multishells/763_1709226474610/bin/node
    npm: 10.2.4 - ~/Library/Caches/fnm_multishells/763_1709226474610/bin/npm
    Watchman: 2024.01.22.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 122.0.6261.94
    Safari: 17.1
  npmPackages:
    @vitejs/plugin-react: ^4.2.1 => 4.2.1 
    @vitest/coverage-v8: ^1.3.1 => 1.3.1 
    vite: ^5.1.4 => 5.1.4 
    vitest: ^1.3.1 => 1.3.1 
```
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by hi-ogawa** [comment]
I cannot reproduce hanging with your simple example.
https://stackblitz.com/edit/vitest-dev-vitest-v6yujf?file=test%2Fbasic.benchmark.ts

It might be possible that some of your plugins might be affecting it. Please provide a reproduction for us to look into the issue further.

Regarding `--reporter=hanging-process`, the benchmark has a different set of reporters and `hanging-process` is not possible to use currently. I would assume `hanging-process` is not a good fit to use with benchmark anyways since its use of `async_hooks` is known to have a significant impact on performance.

**Comment by busticated** [comment]
> Regarding --reporter=hanging-process, the benchmark has a different set of reporters and hanging-process is not possible to use currently.

ok, i assume that's a bug then? presumably we don't want to tell users to use something that is known not to work / help.

**Comment by busticated** [comment]
>It might be possible that some of your plugins might be affecting it.

i didn't realize `defineConfig()` was called with a different `mode` param - though it makes perfect sense that it would be. the fix was for me to add `&& mode !== 'benchmark'` to gate some custom proxy bits 🤦 

thanks for the pointer 🙏👍 

feel free to close this out is the `--reporter=hanging-process` output is being addressed elsewhere

**Comment by hi-ogawa** [comment]
Yeah, benchmark reporter might be confusing at the moment, but I guess it's more like documentation issue since some test reporters cannot be used with benchmark:
https://vitest.dev/guide/reporters.html#basic-reporter
https://vitest.dev/config/#benchmark-reporters

https://github.com/vitest-dev/vitest/blob/c3eb8deb768f6ef1adff8e6644ce99de58fd8b01/packages/vitest/src/node/reporters/index.ts#L30-L41

https://github.com/vitest-dev/vitest/blob/c3eb8deb768f6ef1adff8e6644ce99de58fd8b01/packages/vitest/src/node/reporters/benchmark/index.ts#L5-L9

**Comment by linonetwo** [comment]
https://github.com/vitest-dev/vitest/discussions/4797#discussioncomment-14081107

**Comment by sheremet-va** [comment]
The whole API is being redesigned in https://github.com/vitest-dev/vitest/discussions/7850. Until it's implemented, all benchmark issues are on hold. I recommend following the discussion and giving your feedback there.
